pkg/skaffold/build/tag/sha256.go

```diff
@@ -31,5 +36,17 @@ func (c *ChecksumTagger) Labels() map[string]string {
 }
 
 func (c *ChecksumTagger) GenerateFullyQualifiedImageName(workingDir, imageName string) (string, error) {
-	return imageName, nil
+	// Strip repositories that have a port number.
+	if repoPortRe == nil {
```

Comment:
Please, use something like `parsed, err := docker.ParseReference(tag)` to parse image names.
Also, image names on artifacts should never have a tag. We should maybe enforce that somewhere else.

Reply:
#1623 is where I explained a use case for having tags on built/pushed artifacts.

Thanks so much for your feedback!